### DISCUSSION TITLE
In-depth information about (tiered) profile guided optimization

### DISCUSSION THREAD

**Discussion by trontronicent** [dotnet/runtime#87098] [Q&A] [answered]
Hello fellow DotNetters
I'm looking about in-depth information about (tiered) profile guided optimization within .NET. Anybody knowing a good source of informations about it?

Thanks a lot in advance

All the best from Switzerland
RK

**Comment by stephentoub** [comment, marked as answer]
There's reasonably in-depth coverage in https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-6/ and
https://devblogs.microsoft.com/dotnet/performance_improvements_in_net_7/. Are you looking for something more detailed than that, and if so, in what area specifically?

**Reply by trontronicent** [reply]
Oh wow, thansk a lot Stephentoub. Let me check those first, befor I ask for anything (more). Really appreciated Stephen! A coffee on me ;=)

**Reply by trontronicent** [reply]
Bingo, that's absolutely what I was looking for, Stephen! Very detailed, essentially covering all the question marks in my brain! Thanks again for this!

**Comment by reflectronic** [comment]
There is an overview of (dynamic) PGO in the JIT here: https://github.com/dotnet/runtime/blob/main/docs/design/features/DynamicPgo.md

And a description of the new instrumented tiers for .NET 8 here: https://github.com/dotnet/runtime/blob/main/docs/design/features/DynamicPgo-InstrumentedTiers.md

Is there something more you'd like to see?

**Comment by trontronicent** [comment]
thank you very much reflectronic!  I really appreciate grabbing the link(s) for me!
